docs/src/main/asciidoc/getting-started-reactive.adoc

```diff
@@ -0,0 +1,854 @@
+        .onItem().apply(n -> String.format("hello %s - %d", name, n))
+        .transform().byTakingFirstItems(count);
+}
+----
+
+It generates a greeting message every second and stops after `count` messages.
+
+In the `ReactiveGreetingResource` add the following method:
+
+[source, java]
+----
+@GET
+@Produces(MediaType.APPLICATION_JSON)
+@Path("/greeting/{count}/{name}")
+public Multi<String> greetings(@PathParam int count, @PathParam String name) {
+    return service.greetings(count, name);
+}
+----
+
+This endpoint streams the items to the client as a JSON Array.
+The name and number of messages are parameterized using path parameters.
+
+So calling the endpoint produces something like:
+[source, shell]
+----
+$ curl http://localhost:8080/hello/greeting/3/neo
+["hello neo - 0","hello neo - 1","hello neo - 2"]
+----
+
+We can also generate Server-Sent Event responses by returning a `Multi`:
+
+[source, java]
+----
+@GET
+@Produces(MediaType.SERVER_SENT_EVENTS)
+@SseElementType(MediaType.TEXT_PLAIN)
+@Path("/stream/{count}/{name}")
+public Multi<String> greetingsAsStream(@PathParam int count, @PathParam String name) {
+    return service.greetings(count, name);
+}
+----
+
+The only difference with the previous snippet is the produced type and the `@SseElementType` annotation indicating the type of each event.
+
+You can see the result using:
+
+[source, shell]
+----
+$ curl -N http://localhost:8080/hello/stream/5/neo
+data: hello neo - 0
+
+data: hello neo - 1
+
+data: hello neo - 2
+
+data: hello neo - 3
+
+data: hello neo - 4
+----
+
+== Using Reactive APIs
+
+=== Using Quarkus reactive APIs
+
+Quarkus provides many reactive APIs using the Mutiny model.
+In this section, we are going to see how you can use the Reactive PostgreSQL driver to interact with your database in a non-blocking and reactive way.
+
+Create a new project using:
+
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=getting-started-reactive-crud \
+    -DclassName="org.acme.reactive.crud.FruitResource" \
+    -Dpath="/fruits" \
+    -Dextensions="resteasy-mutiny, resteasy-jsonb, reactive-pg-client"
+cd getting-started-reactive-crud
+----
+
+This application is interacting with a Postgresql database, so you need one:
+
+[source, shell]
+----
+docker run --ulimit memlock=-1:-1 -it --rm=true --memory-swappiness=0 \
+           --name postgres-quarkus-reactive -e POSTGRES_USER=quarkus_test \
+           -e POSTGRES_PASSWORD=quarkus_test -e POSTGRES_DB=quarkus_test \
+           -p 5432:5432 postgres:10.5
+----
+
+Then, let's configure our data source.
+Open the `src/main/resources/application.properties` and add the following content:
+
+[source, properties]
+----
+quarkus.datasource.url=vertx-reactive:postgresql://localhost:5432/quarkus_test
+quarkus.datasource.username=quarkus_test
+quarkus.datasource.password=quarkus_test
```

Comment:
```suggestion
quarkus.datasource.db-kind=postgresql
quarkus.datasource.username=quarkus_test
quarkus.datasource.password=quarkus_test
quarkus.datasource.reactive.url=vertx-reactive:postgresql://localhost:5432/quarkus_test
```
would be the new way of doing things now.

Reply:
IIRC, you can even remove the `vertx-reactive:` part when doing it this way.

Reply:
I will try tomorrow morning.

Reply:
It's working, thanks!